src/Sylius/Bundle/ApiBundle/Validator/Constraints/PromotionCouponEligibility.php

```diff
@@ -17,8 +17,11 @@
 
 final class PromotionCouponEligibility extends Constraint
 {
-    /** @var string */
-    public $message = 'sylius.promotion_coupon.is_invalid';
+    public string $invalid = 'sylius.promotion_coupon.is_invalid';
+
+    public string $expired = 'sylius.promotion_coupon.expired';
+
+    public string $ineligible = 'sylius.promotion_coupon.ineligible';
```

Comment:
```suggestion
    public string $invalidMessage = 'sylius.promotion_coupon.is_invalid';

    public string $expiredMessage = 'sylius.promotion_coupon.expired';

    public string $ineligibleMessage = 'sylius.promotion_coupon.ineligible';
```

Reply:
Those `public` variables are accessible from `validation.yaml` configuration, where you can setup custom message, so it should be named `$acmeMessage`.